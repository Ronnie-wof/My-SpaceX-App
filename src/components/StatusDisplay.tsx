// Rocket status display component (currently displayed on RocketInfoScreen)
import React from 'react'
import {StyleSheet, Text} from 'react-native'
import { AntDesign } from '@expo/vector-icons'; 
import { Entypo } from '@expo/vector-icons';

interface statusDisplayProps {
    label: string,
    status: boolean | null
}

const StatusDisplay:React.FC<statusDisplayProps> = ({label, status}) => {
    return (
        <Text style={styles.small}>
                {label}: { status == null && <Text style={styles.text}>No data available</Text> } 
                {status === true && <AntDesign name="checkcircle" size={24} color="black" />}
                {status === false && <Entypo name="circle-with-cross" size={24} color="black"  />}
        </Text>
    );
}

const styles = StyleSheet.create({
    small: {
        fontSize: 18,
        fontWeight: 'bold',
        fontStyle: 'italic'
    },
    text: {
        fontWeight: 'normal',
        fontStyle: 'italic'
    }
});

export default StatusDisplay;