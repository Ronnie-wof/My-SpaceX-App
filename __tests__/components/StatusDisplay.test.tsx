// tests the StatusDisplay component
import React from 'react';
import renderer from 'react-test-renderer';
import StatusDisplay from '../../src/components/StatusDisplay';

describe('StatusDisplay component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<StatusDisplay label="Recovered" status={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
