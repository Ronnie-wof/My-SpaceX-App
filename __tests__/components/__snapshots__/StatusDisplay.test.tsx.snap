// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusDisplay component renders correctly 1`] = `
<Text
  style={
    {
      "fontSize": 18,
      "fontStyle": "italic",
      "fontWeight": "bold",
    }
  }
>
  Recovered
  : 
  <Text />
</Text>
`;
